internal/provider/datasource_cloud_locations.go

```diff
@@ -0,0 +1,262 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/PacketFabric/terraform-provider-packetfabric/internal/packetfabric"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func datasourceCloudLoctions() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceCloudLocationsRead,
+		Schema: map[string]*schema.Schema{
+			"cloud_provider": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"aws", "azure", "packet", "google", "ibm", "oracle", "salesforce", "webex"}, true),
+				Description:  "Filter locations by cloud provider.",
+			},
+			"cloud_connection_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"hosted", "dedicated"}, true),
+				Description:  "Filter locations by cloud connection type.",
+			},
+			"cloud_locations": {
+				Type:     schema.TypeList,
```

Comment:
@pontovinte @elXaptation could we add a description for cloud_locations?

Something like `Description: "The list of list of physical locations optionally filtered by provided parameters.",`